src/message-io.js

```diff
@@ -85,18 +85,29 @@ module.exports = class MessageIO extends EventEmitter {
     return this._packetSize;
   }
 
-  startTls(credentialsDetails) {
+  startTls(credentialsDetails, trustServerCertificate) {
     const credentials = tls.createSecureContext ? tls.createSecureContext(credentialsDetails) : crypto.createCredentials(credentialsDetails);
 
     this.securePair = tls.createSecurePair(credentials);
     this.tlsNegotiationComplete = false;
 
-    this.securePair.on('secure', () => {
-      const cipher = this.securePair.cleartext.getCipher();
-      this.debug.log('TLS negotiated (' + cipher.name + ', ' + cipher.version + ')');
-      this.emit('secure', this.securePair.cleartext);
-      this.encryptAllFutureTraffic();
-    });
+    this.securePair.on('secure', (_this) => {
```

Comment:
this.ssl comes from the callback function of this.securePair.on( 'secure' )

To get access to `this.ssl.verifyError` I needed access to the `this` scope from the `secure` event so I needed to change away from an arrow function but still needed access to the parent `this` scope to access the other methods such as `securePair` and `encryptAllFutureTraffic` as they are not defined in the "this" scope of the 'secure' event.